storage/engine/mvcc.go

```diff
@@ -956,23 +956,125 @@ func MVCCMerge(engine Engine, ms *MVCCStats, key roachpb.Key, value roachpb.Valu
 
 // MVCCDeleteRange deletes the range of key/value pairs specified by
 // start and end keys. Specify max=0 for unbounded deletes.
-func MVCCDeleteRange(engine Engine, ms *MVCCStats, key, endKey roachpb.Key, max int64, timestamp roachpb.Timestamp, txn *roachpb.Transaction) (int64, error) {
-	// In order to detect the potential write intent by another
-	// concurrent transaction with a newer timestamp, we need
-	// to use the max timestamp for scan.
-	kvs, _, err := MVCCScan(engine, key, endKey, max, roachpb.MaxTimestamp, true /* consistent */, txn)
-	if err != nil {
-		return 0, err
+func MVCCDeleteRange(engine Engine, ms *MVCCStats, startKey, endKey roachpb.Key, max int64, timestamp roachpb.Timestamp, txn *roachpb.Transaction) (int64, error) {
+	if len(endKey) == 0 {
+		return 0, emptyKeyError()
 	}
 
-	num := int64(0)
-	for _, kv := range kvs {
-		if err := MVCCDelete(engine, ms, kv.Key, timestamp, txn); err != nil {
-			return num, err
+	buf := putBufferPool.Get().(*putBuffer)
+	defer putBufferPool.Put(buf)
+	buf.value.Reset()
+	buf.value.Deleted = true
+
+	// We store encEndKey and encKey in the same buffer to avoid memory
+	// allocations.
+	encEndKey := mvccEncodeKey(buf.key[0:0], endKey)
```

Comment:
True, but does it matter? The generated code will be the same. I'll change the code this time, but I don't think we should be nit picking about such items.